src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Internal/MsQuicSession.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Net.Security;
+using System.Text;
+
+namespace System.Net.Quic.Implementations.MsQuic.Internal
+{
+    internal sealed class MsQuicSession : IDisposable
+    {
+        private bool _disposed = false;
+        private IntPtr _nativeObjPtr;
+        private MsQuicApi _api;
+        private bool _opened;
+
+        internal MsQuicSession(string registration)
+        {
+            _api = new MsQuicApi();
+
+            _api.RegistrationOpen(Encoding.UTF8.GetBytes(registration));
```

Comment:
As mentioned before, I'm not a fan of this abstraction layer merging Api, Registration and Session into a single object. If you want a single object that wraps them, I'd prefer to have a layer on top of a MsQuicApi, MsQuicRegistration and MsQuicSession objects.